bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java

```diff
@@ -798,7 +835,7 @@ public void openComplete(int rc, final LedgerHandle lh, Object ctx) {
                         @Override
                         public void openComplete(int newrc, final LedgerHandle newlh, Object newctx) {
                             if (newrc != BKException.Code.OK) {
-                                LOG.error("BK error close ledger: " + lId, BKException.create(newrc));
+                                LOG.error("BK error close ledger: {}", lId, BKException.create(newrc));
                                 finalLedgerIterCb.processResult(newrc, null, null);
```

Comment:
Do we need to complete the `finalLedgerIterCb` normally? In the recover ledger method, I saw multiple places still failed the `finalLedgerIterCb` and it will break the recover process. Should we handle them?

Reply:
Thanks for you reply @zymap. You are right,  I update the code to handle them, Please help take a look, thanks.